storm-client/src/jvm/org/apache/storm/cluster/StormClusterStateImpl.java

```diff
@@ -438,9 +438,7 @@ public void workerBackpressure(String stormId, String node, Long port, long time
         String path = ClusterUtils.backpressurePath(stormId, node, port);
         boolean existed = stateStorage.node_exists(path, false);
         if (existed) {
-            if (timestamp <= 0) {
-                stateStorage.delete_node(path);
-            } else {
+            if (timestamp > 0) {
```

Comment:
The change seems reasonable to me. The javadoc for this method is slightly out of date now, consider updating the first line of it.

The two outermost branches here do mostly the same thing. Consider rewriting to something like 
```
if (timestamp <= 0) {
  return
}
String path = ...
boolean existed = ...
byte[] data = ...
if (existed) {
  set_data
} else {
  set_ephemeral_node
}
```